mocks/producer.go

```diff
@@ -0,0 +1,138 @@
+package mocks
+
+import (
+	"sync"
+
+	"github.com/Shopify/sarama"
+)
+
+// Producer implements sarama's Producer interface for testing purposes.
+// Before you can send messages to it's Input channel, you have to set expectations
+// so it knows how to handle the input. This way you can easily test success and
+// failure scenarios.
+//
+// NOTE: the SyncProducer type currently does not fall under the API stability
+// guarantee of Sarama as it is stiull considered experimental.
```

Comment:
I would put the note about api stability at the package level, not on individual types